### ISSUE TITLE
TypeScript Syntax Highlighting Bug

### ISSUE THREAD

**Issue by fc01** [microsoft/TypeScript#34517]
```ts
//正常的颜色
const f0 = <T extends any>(p: {
    a: string
    b: T
}) => async () => p.a + p.b


//颜色不对
const f1 = <T extends any>({ a, b }: {
    a: string
    b: T
}) => async () => a + b


//颜色不对
const f2 = <T extends any>({ a, b }: {
    a: string
    b: T
}) => async () => a + b
```
 
![image](https://user-images.githubusercontent.com/9409898/66941808-4e30ad00-f07a-11e9-9890-3d658db79780.png)

**Comment by AnyhowStep** [comment]
A TS problem or a VS code problem?

**Comment by zw627** [comment]
Did some tests. (VS Code 1.39.2, TypeScript 3.6.3.)

Same problem, but it seems to be fixed after running through the formatter:
![one-dark-pro-ts](https://user-images.githubusercontent.com/31707422/66950110-727f9000-f060-11e9-866d-148de3b58851.png)

TSX seems to work just fine:
![one-dark-pro-tsx](https://user-images.githubusercontent.com/31707422/66950115-74e1ea00-f060-11e9-8f62-b8e49e12fe06.png)

**Comment by AnyhowStep** [comment]
Might possibly be related to https://github.com/microsoft/vscode/issues/77140 ?

**Comment by mjbvz** [comment]
This was also reported against VS Code. Moved here: https://github.com/microsoft/TypeScript-TmLanguage/issues/780

**Comment by inspiredjw** [comment]
https://medium.com/@inspiredjw/fix-vscode-syntax-highlighting-with-typescript-3-7-3dfc79db0c7a
